### DISCUSSION TITLE
I want to opt-out of my data being used for algorithmic feeds

### DISCUSSION THREAD

**Discussion by stagas** [community/community#13162] [Other Feature Feedback, Questions, & Ideas]
As a GitHub user I would like to use the site without my activites being included in any weighted algorithm that modifies views for me or anyone else. How can I opt-out of all weighted algorithmic side-effects?

**Comment by bdresser** [comment]
Hi @stagas – we're working on a feature that will allow you to make your profile "private" and opt out from Trending, For you, Discussions leaderboards, your Profile activity feed, and Following all in one setting. We're aiming to launch in late April. 

If you have other suggestions for how you want (or don't want) your activity aggregated across GitHub, feel free to reach out: bdresser@github.com

**Reply by phikal** [reply]
Why is this opt-out instead of opt-in?

**Reply by adriano66** [reply]
> Why is this opt-out instead of opt-in?

Because "dark patters" that's why

**Reply by suprstarrd** [reply]
I hope this gets across properly. Writing this as fast as I can.

For me, this doesn't solve the underlying problem: This isn't needed! I don't want to be tracked on GitHub.

Having "For You" just feels... *creepy*. Knowing the things I like so specifically because of the repositories I visit on GitHub feels like a boundary has been broken. Despite that, I can socialize and participate: making issues and pull requests, leaving comments, following a user, and starring their repository; these are all things I can do without data collection! Source: I have `collector.github.com` blocked in my browser.

I know that despite my best efforts, I'm probably going to have data collected on my GitHub user and the things that happen on it, ["anonymous"](https://spreadprivacy.com/data-anonymization/) or not. But I do my best to limit it. In fact, most of the things I make are made to protect your privacy. And I do all of this *on GitHub*. I have the [Student Developer Pack](https://education.github.com/pack), and one of my online friends is trying to get their school enrolled in the [GitHub Campus Program](https://education.github.com/schools). We work together all the time, and that's only possible because of GitHub. It's genuinely helped us make things.

I really, really, *really* truly love GitHub, and I won't leave even if this change does happen. *However*, I will be seriously disappointed if this leaves beta without a true opt-out option. It hurts more knowing this "private" mode likely won't truly be private.

I know my feedback probably will be disregarded but ugh. This is a bad compromise.

**Reply by ghost** [reply]
> Why is this opt-out instead of opt-in?

Why are you even expecting an Commercial service owned by Microsoft to be Private Lol.

**Reply by OdinVex** [reply]
Just another reason to self-host. Gitea/Gogs, decent alternatives, _especially_ for GitHub free users. We knew it'd go to `....` once Micro`....` bought GitHub. Just a matter of time, always is.

**Reply by fbnkrs** [reply]
Stop this nonsense.

Noone wants those "features".
This is highly unprofessional behaviour.

**Reply by ghost** [reply]
> Stop this nonsense.
> 
> Noone wants those "features". This is highly unprofessional behaviour.

Agreed.

**Reply by mtimofiiv** [reply]
> we're working on a feature that will allow you to make your profile "private" and opt out

Call me cynical but I feel like there will be an opt-out until a certain day when you just remove the opt-out entirely and we're stuck with this algorithmic garbage with no choice.

**Reply by TeoTwawki** [reply]
> Hi @stagas – we're working on a feature that will allow you to make your profile "private" and opt out from Trending, For you, Discussions leaderboards, your Profile activity feed, and Following all in one setting. We're aiming to launch in late April.
> 
> If you have other suggestions for how you want (or don't want) your activity aggregated across GitHub, feel free to reach out: [bdresser@github.com](mailto:bdresser@github.com)

@bdresser I don't want to make my profile private - that implies more than you are describing. I just want github to not use it for garbage I don't approve of. I'm on github, not facebook. If github keeps trying to make itself more like facebook I'll be leaving, all my projects will be leaving, all the teams I have any authority in will be leaving and I don't think anyone would even disagree with me on it because none of us want this logarithmic feed nonsense. Period.

**Comment by stagas** [comment]
@bdresser "private" is a good feature to have as well, but the reason i'm using GitHub is precisely because of its social activity, I simply don't want that activity to carry algorithmic side-effects. In essence, I'd like be able to star someone's project without that necessarily meaning that now I want to stalk all of their activity all of the time or that I'll be included in their views more prominently because of that star. I'd like my actions to be free of weighted algorithmic side-effects, esp. those that are personalized. The global trends and effects that are not personalized are ok, i guess, an option would be great there as well. So my request is not the same as becoming "private". I still want to keep all activity flowing normally in a chronological manner, just without any algorithm modifying it. Is it possible to consider such a feature or it can only be all or nothing?

**Comment by jfullstackdev** [comment]
> but the reason i'm using GitHub is precisely because of its social activity, I simply don't want that activity to carry algorithmic side-effects.

being private is already good 

but maybe the request here is even if you made your profile private, much like Twitter, followers can still see your activity, so you can select who can see your activities
